pkg/network/netlink/socket.go

```diff
@@ -201,56 +201,34 @@ func (s *Socket) ReceiveAndDiscard() (bool, uint32, error) {
 // ReceiveInto reads one or more netlink.Messages off the socket
 func (s *Socket) ReceiveInto(b []byte) ([]netlink.Message, uint32, error) {
 	var netns uint32
-	var ret []netlink.Message
-	for {
-		n, oobn, err := s.recvmsg()
-		if err != nil {
-			return nil, 0, os.NewSyscallError("recvmsg", err)
-		}
+	n, oobn, err := s.recvmsg()
+	if err != nil {
+		return nil, 0, os.NewSyscallError("recvmsg", err)
+	}
 
-		n = nlmsgAlign(n)
-		// If we cannot fit the date into the supplied buffer,  we allocate a slice
-		// with enough capacity. This should happen very rarely.
-		if n > len(b) {
-			b = make([]byte, n)
-		}
-		copy(b, s.recvbuf[:n])
+	n = nlmsgAlign(n)
+	// If we cannot fit the date into the supplied buffer,  we allocate a slice
+	// with enough capacity. This should happen very rarely.
+	if n > len(b) {
+		b = make([]byte, n)
+	}
+	copy(b, s.recvbuf[:n])
+
+	msgs, err := ParseNetlinkMessage(b[:n])
+	if err != nil {
+		return nil, 0, err
+	}
 
-		msgs, err := ParseNetlinkMessage(b[:n])
+	if oobn > 0 {
+		scms, err := unix.ParseSocketControlMessage(s.oobbuf[:oobn])
 		if err != nil {
 			return nil, 0, err
 		}
 
-		var multi bool
-		for _, m := range msgs {
-			if err := checkMessage(m); err != nil {
```

Comment:
Is this no longer relevant?

Reply:
Happens at the caller level: https://github.com/DataDog/datadog-agent/blob/9f6ec93ec61408c43c66ac4d2b4d89f31955cedd/pkg/network/netlink/consumer.go#L545, so not needed here.